static/js/settings_bots.js

```diff
@@ -291,6 +291,25 @@ export function add_a_new_bot() {
         });
     }
 
+    function validate_input(e) {
+        e.preventDefault();
+        e.stopPropagation();
+
+        const bot_short_name = $("#create_bot_short_name").val();
+
+        if (/^[\w!#$%&'*+/=?^`{|}~-]+(\.[\w!#$%&'*+/=?^`{|}~-]+)*$/i.test(bot_short_name)) {
+            return true;
+        }
```

Comment:
What's the source for this regular expression? I imagine you got it from StackOverflow or something; if so, we should make a function `is_valid_email_address_username` that does that check, and cite the source for the code in a comment in that function.

Reply:
I see, it was present in a previous commit but deleted, including the comment explaining where it came from. Having re-adding this be a separate commit from the commit where you removed that validation was very confusing.

Reply:
```
-function is_local_part(value, element) {
-    // Adapted from Django's EmailValidator
-    return (
-        this.optional(element) ||
-        /^[\w!#$%&'*+/=?^`{|}~-]+(\.[\w!#$%&'*+/=?^`{|}~-]+)*$/i.test(value)
-    );
-}
```

Reply:
I will take care of the comments next time.